include/deal.II/distributed/grid_tools.h

```diff
@@ -134,6 +134,60 @@ namespace parallel
     exchange_cell_data_to_ghosts (const MeshType &mesh,
                                   const std::function<boost::optional<DataType> (const typename MeshType::active_cell_iterator &)> &pack,
                                   const std::function<void (const typename MeshType::active_cell_iterator &, const DataType &)> &unpack);
+
+    /**
+     * A structure that allows the transfer of data of type T from one processor
+     * to another. It corresponds to a packed buffer that stores a list of
+     * cells and an array of type T.
+     *
+     * The vector @p data is the same size as @p cell_ids.
+     */
+    template <int dim, typename T>
+    struct CellDataTransferBuffer
```

Comment:
* I think it would be useful if the class doc would say that the class facilitates this transfer by providing the `save`/`load` functions that are able to pack everything up into a stream.

* I think it would also be useful to state the requirements on `T`, namely that it can be serialized via the `boost::serialization` framework.